python_modules/dagster/dagster/_core/definitions/schedule_definition.py

```diff
@@ -291,6 +299,10 @@ def has_captured_logs(self):
     def log_key(self) -> Optional[List[str]]:
         return self._log_key
 
+    @property
+    def repository_def(self) -> Optional["RepositoryDefinition"]:
```

Comment:
The pattern we generally use elsewhere with context methods is to make the return types non-optional - e.g. with the `instance` and `scheduled_execution_time` properties. The advantage of this is that, when someone is using them in a normal way, they don't need to put up null guards to make their code work with type checking.

Reply:
good call